feat:implememtation  attendance

Patienceineza opened the pull request:
# PR Description
Attendance
a coordinator should see the list of attendance
 
- **attendance distribution**
     - 0: Didn't attend and didn't communicate
     - 1: Didn't attend but communicated
     - 2: attended
 
Attendance statistics
 
# How has this been tested?
- For browser
  - Go [Here](https://metron-devpulse-git-ft-attendance-metron.vercel.app) and login as Coordinator
     - email: alineuwera00@gmail.com
     - password: Test@12345
     
  - click on attendance you will be able to see attendance based on week and corresponding statistics
  - To record attendance follow this:
     - after click on attendance click on submit attendance.
     - choose week, day and assign value to trainnee then save
# Number of Commits

1

# Screenshots (If appropriate)
<img width="797" alt="attendance" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112135715/19f6f2e6-cbac-4f35-b7b8-59a4e6f5671e">

# Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

blackd44 reviewed (commented):
other are fine
check your pagination
<img width="1438" alt="Screenshot 2023-08-07 at 21 23 13" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/bc502603-b817-4b69-9e65-a3340a21b948">

ceelogre reviewed (commented):
The trainees' list in calendar is empty yet there are some visible on the Trainee's page.
<img width="655" alt="image" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/5022823/21f36b83-1b90-430c-9109-71ab5df17284">
<img width="645" alt="image" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/5022823/d7913278-419e-4ac6-a287-7d074559e136">

ceelogre commented:
Why is it that the trainee list is empty
![image](https://github.com/atlp-rwanda/atlp-pulse-fn/assets/5022823/9ec8fbd8-5f81-4d24-9f26-239e71566014)

yet there appears to be some trainees in attendance?
![image](https://github.com/atlp-rwanda/atlp-pulse-fn/assets/5022823/24296458-8f26-4d99-96f3-46cdee65ba01)

UwaweTera committed:
assign attendance value to trainees

- design  the help page form
- validate the user for help page
-create the ticket for superadmin for receiveing the feedback

[Delviers #28-help-page]

feat(help-page): submit page

 - ensure users can send messages to superadmin
 - create create help mutation

[Deliver #ft-28-help-page]

 - ensure that superadmin can read messages
 - create nav link on sidebar of superadmin dashboard
 - display all messageds

[Deliver #ft-28-help-page]

feat(help page): pagination

 - add button based pagination

[Deliver #ft-28-help-page]

bug(help page): fix toast notification

 - only show toast notification when user successfully submits message

[Fixes #ft-28-help-page]

fix response on toast notification upon successful submision of message

feat(help page): clear form ufter submission

feat(ticket system): create context

 - ensure that tickets can be availble anywhere in the app
 - create context for tickets

[Deliver #39-ticket-system-continued]

feat(ticket system): view tiket

 - ensure users can read details of a ticket
 - create ticket page to display details of a ticket

[Deliver #39-ticket-system-continued]

feat(ticket system): replies

 - ensure users can see remplies on tickey detail page
 - build templates for replies on ticket deetail page

[Delivers #39-ticket-system-continued]

Co-authored-by: Mystica <r.dukuzeyez@alustudent.com>

* feat(login):user can see login activities

* fix: remove pnpm lock

* tsconfig.json fix

---------

Co-authored-by: ceelogre <ceelogre@outlook.com>
Co-authored-by: newby101 <ishimwekevin45@gmail.com>

- design  the help page form
- validate the user for help page
-create the ticket for superadmin for receiveing the feedback

[Delviers #28-help-page]

feat(help-page): submit page

 - ensure users can send messages to superadmin
 - create create help mutation

[Deliver #ft-28-help-page]

feat(help page): display messages

 - ensure that superadmin can read messages
 - create nav link on sidebar of superadmin dashboard
 - display all messageds

[Deliver #ft-28-help-page]

feat(help page): pagination

 - add button based pagination

[Deliver #ft-28-help-page]

bug(help page): fix toast notification

 - only show toast notification when user successfully submits message

[Fixes #ft-28-help-page]

fix response on toast notification upon successful submision of message

feat(help page): clear form ufter submission

feat(ticket system): create context

 - ensure that tickets can be availble anywhere in the app
 - create context for tickets

[Deliver #39-ticket-system-continued]

feat(ticket system): view tiket

 - ensure users can read details of a ticket
 - create ticket page to display details of a ticket

[Deliver #39-ticket-system-continued]

feat(ticket system): replies

 - ensure users can see remplies on tickey detail page
 - build templates for replies on ticket deetail page

[Delivers #39-ticket-system-continued]

Co-authored-by: Mystica <r.dukuzeyez@alustudent.com>

* feat(login):user can see login activities

* fix: remove pnpm lock

* tsconfig.json fix

---------

Co-authored-by: ceelogre <ceelogre@outlook.com>
Co-authored-by: newby101 <ishimwekevin45@gmail.com>

assign attendance  value to trainee

assign attendance value to trainees
